py_ecc/validate_constants.py

```diff
@@ -0,0 +1,135 @@
+from functools import lru_cache
+
+from py_ecc.bls12_381_curve import (
+    bls12_381,
+    optimized_bls12_381,
+)
+
+from py_ecc.bn128_curve import (
+    bn128,
+    optimized_bn128,
+)
+
+from py_ecc.curve_properties import (
+    curve_properties,
+    optimized_curve_properties,
+)
+
+from py_ecc.field_properties import (
+    field_properties,
+)
+
+
+def validate_field_properties() -> None:
+    for curve_name in field_properties:
+        # Check if field_modulus is prime
+        field_modulus = field_properties[curve_name]["field_modulus"]
+        if pow(2, field_modulus, field_modulus) != 2:
```

Comment:
I feel dense.  How does this check for *primeness*?

Reply:
@pipermerriam I believe that is the `Fermat's Little Theorem`.

Reply:
If I'm reading this correctly, the test correctly detects **some, but not all** cases where `field_modulus` is not prime. Maybe the comment could be updated since this is kind of a "smoke test" rather than a proof of primeness.

(For example, see `341` which is not prime, but would not raise this exception)

Reply:
@carver nice catch. Didn't realize that. Will fallback to the traditional `O(sqrt(n))` way of checking a prime number. Thankyou!

Reply:
Since the `field_modulus` and the other numbers that we are checking for primes, are quite large, the `O(sqrt(n))` method is taking a lot of time. We might as well stay with the present `smoke test`.